torch/lib/THD/base/ChannelUtils.hpp

```diff
@@ -192,4 +187,22 @@ void send_value(int socket, T&& value, bool more_data = false) {
   send_bytes<T>(socket, &value, 1, more_data);
 }
 
+class ResourceGuard {
```

Comment:
It's on purpose. It seems cleaner for me to use structs for simple records that have no methods, and class for objects